lib/namespace-hmac.js

```diff
@@ -0,0 +1,287 @@
+/*!
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const _ = require('lodash');
+const async = require('async');
+const bedrock = require('bedrock');
+const crypto = require('crypto');
+const database = require('bedrock-mongodb');
+const jwt = require('jsonwebtoken');
+const BedrockError = bedrock.util.BedrockError;
+const Namespace = require('./namespace');
+
+const logger = bedrock.loggers.get('app');
+
+class NamespaceHmac extends Namespace {
+  constructor() {
+    super();
+    this._keyState = null;
+  }
+
+  createState(options, callback) {
+    const self = this;
```

Comment:
Is the frequent use of the `self` alias vs plain `this` just for consistency?  The `self` pattern is useful to access an outer `this` from an inner function but many of these uses are on simple functions or ones with inner arrow functions where `self` isn't really needed.